include/accessor.h

```diff
@@ -80,20 +89,27 @@ class host_memory_layout {
 		size_t extent{};
 	};
 
-	/** Since contiguous dimensions can be merged when generating the memory layout, host_memory_layout is not generic over a fixed dimension count */
-	constexpr static size_t max_dimensionality = 4;
-
-	using dimension_vector = boost::container::static_vector<dimension, max_dimensionality>;
-
-	explicit host_memory_layout(const dimension_vector& dimensions) : dimensions(dimensions) {}
+	size_t get_dimensions() const { return dimensions.size(); }
```

Comment:
This should be `get_num_dimensions` or `get_num_strides` depending on how we want to name the type (it returns a count rather than a sequence)